tket/src/Circuit/macro_manipulation.cpp

```diff
@@ -708,10 +708,19 @@ bool Circuit::substitute_box_vertex(
   return true;
 }
 
-bool Circuit::decompose_boxes() {
+bool Circuit::decompose_boxes_recursively(
```

Comment:
We're using `std::list` to store the graph vertices. Also, according to the Boost documentation, vertices are added to the graph using `std::list::push_back`. So, any vertex added inside the loop will always get looped through in the future. `BGL_FORALL_VERTICES` essentially just goes through the `std::list`, so the code here shouldn't cause any undefined behaviour.